platform/ios/app/MBXStateManager.m

```diff
@@ -0,0 +1,43 @@
+#import "MBXStateManager.h"
+#import <Mapbox/Mapbox.h>
+#import "MBXState.h"
+#import "MBXViewController.h"
+
+@interface MBXStateManager()
+
+@property (strong, nonatomic) MBXState *currentState;
+
+@end
+
+@implementation MBXStateManager
+
++ (instancetype) sharedManager {
+    static dispatch_once_t once;
+    static MBXStateManager* sharedManager;
+    dispatch_once(&once, ^{
+        sharedManager = [[self alloc] init];
+    });
+
+    return sharedManager;
+}
+
+- (MBXState*)currentState {
+    NSData *encodedMapState = [[NSUserDefaults standardUserDefaults] objectForKey:@"mapStateKey"];
```

Comment:
For a stringly-typed key name that’s used multiple times, it’s a good practice to make it a constant, so that all uses remain in sync.